src/DotNetLightning.Core/Channel/CommitmentsModule.fs

```diff
@@ -497,47 +497,57 @@ module RemoteForceClose =
             }
             |> Seq.sortWith TxOut.LexicographicCompare
 
-        let toRemoteIndex =
+        let toRemoteIndexOpt =
             outputs
-            |> Seq.findIndex (fun out -> out.ScriptPubKey = toRemoteScriptPubKey)
-
-        let toRemoteTxOut: TxOut = 
-            outputs 
-            |> Seq.item toRemoteIndex
-
-        let localPaymentPrivKey =
-            perCommitmentPoint.DerivePaymentPrivKey localChannelPrivKeys.PaymentBasepointSecret
-
-        (transactionBuilder.AddKeys(localPaymentPrivKey.RawKey()))
-            .AddCoins(Coin
-                          (commitments.RemoteCommit.TxId.Value,
-                           toRemoteIndex |> uint32,
-                           toRemoteTxOut.Value,
-                           toRemoteTxOut.ScriptPubKey))
-        |> ignore
+            |> Seq.tryFindIndex (fun out -> out.ScriptPubKey = toRemoteScriptPubKey)
 
-        let toLocalIndex =
+        let _spentToRemoteOutput = 
+            match toRemoteIndexOpt with
+                | None -> None
+                | Some toRemoteIndex ->
+                    let toRemoteTxOut: TxOut = 
```

Comment:
Rather than create the `TxOut` above in the `seq` expression and then using `item` to retrieve it here, why not create it separately then pass it into the `seq`? That would have prevented the bug that made this commit necessary. Fallible methods like `item` should be avoided where simple code-restructuring makes them unnecessary. (I was actually gonna comment on this in the last commit but I didn't want to be too nit-picky and I didn't realize it was an actual bug rather than just a bug-prone way to write the code).